mlir/lib/Analysis/Presburger/Matrix.cpp

```diff
@@ -432,4 +432,138 @@ MPInt IntMatrix::normalizeRow(unsigned row, unsigned cols) {
 
 MPInt IntMatrix::normalizeRow(unsigned row) {
   return normalizeRow(row, getNumColumns());
+}
+
+MPInt IntMatrix::determinant() {
+  unsigned r = getNumRows();
+  unsigned c = getNumColumns();
+  assert(r == c);
+
+  FracMatrix m(r, c);
+  for (unsigned i = 0; i < r; i++)
+    for (unsigned j = 0; j < c; j++)
+      m.at(i, j) = Fraction(at(i, j), 1);
```

Comment:
isn't there an assignment operator from MPInt to Fraction?

Reply:
it's ambiguous because both
`Fraction(const MPInt &oNum, const MPInt &oDen = MPInt(1))`
and
`Fraction(const MPInt &num, int64_t den = 1) : Fraction(num, MPInt(den))`
apply. should I make one of these not have a default second argument?

Reply:
Fixed upstream